spec/models/user_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Koshao', photo: 'https://avatars.githubusercontent.com/u/92449654?v=4', post_counter: 10, bio: 'Software Developer') }
+
+  before { subject.save }
+
+  it 'Name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Post counter must be an integer' do
+    subject.post_counter = 'z'
+    expect(subject).to_not be_valid
+  end
+
+  it 'Post counter should be 0 or greater' do
+    subject.post_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
+
+RSpec.describe User, type: :model do
+  describe '#recent_posts' do
+    let(:new_user) { User.create(name: 'Kunda', photo: 'https://images.unsplash.com/photo-1633332755192-727a05c4013d', bio: 'Tella') }
+
+    it 'returns zero(0) when the user is created' do
+      expect(new_user.recent_posts.count).to eq 0
+    end
```

Comment:
- [ ] Just as the previous reviewer highlighted, the test is still failing on this end here. Kindly work on it and be sure it passes before requesting a review 👍🏽

<img width="713" alt="Screenshot 2023-06-10 at 12 12 06 PM" src="https://github.com/Koshao-sys/blog-app/assets/50941074/438c5e76-eb80-4756-9f9d-ec5b7b2bc57c">